### PR TITLE
Add incident and progress DB table

### LINKED ISSUE #929: [Feature] Add migration and SQL files for Incident and Progress data model
**What is the problem?**
We need to add migration and SQL for the Incident and Progress data model

image

**Your solution**
I will add migration and SQL files for the Incident and Progress data model based on our DB Schema

### PR DESCRIPTION
**Pull request by Tlac**
### Description
Fixes #929. Adds the migration and SQL files for the Incident and Progress data models based on the UML. 
- The incident table will be used to store various incidents that are reported such as something that causes downtime or reduced performance. 
- The progress table will be used to store the current and past progress of the incidents.

### Screenshots
image

### CODE REVIEW

File: backend/app/adapter/sqldb/table/incident.go [hunk 1]
```diff
@@ -0,0 +1,14 @@
+package table
+
+// Incident represents database table columns for 'incident' table
+var Incident = struct {
+  TableName       string
+  ColumnID        string
```
Comment (arberiii) [inline backend/app/adapter/sqldb/table/incident.go:6]: Please run the ./backend/scripts/format, to format the code

File: backend/app/adapter/sqldb/table/progress.go [hunk 2]
```diff
@@ -0,0 +1,16 @@
+package table
+
+// Progress represents database table columns for 'progress' table
+var Progress = struct {
+  TableName           string
+  ColumnIncidentId    string
```
Comment (arberiii) [inline backend/app/adapter/sqldb/table/progress.go:6]:
```suggestion
  TableName           string
  ColumnIncidentID    string
```

File: backend/app/adapter/sqldb/migration/29_create_incident_table.sql [hunk 3]
```diff
@@ -0,0 +1,10 @@
+-- +migrate Up
+CREATE TABLE incident
+(
+    id          CHARACTER VARYING(4)      NOT NULL UNIQUE,
```
Comment (byliuyang) [inline backend/app/adapter/sqldb/migration/29_create_incident_table.sql:4]:
```suggestion
    id          VARCHAR(4) NOT NULL PRIMARY KEY,
```
id is the only way to uniquely `INDENTIFY` a incident. So we should it as the `PRIMARY KEY`.

File: backend/app/adapter/sqldb/migration/29_create_incident_table.sql [hunk 4]
```diff
@@ -0,0 +1,10 @@
...
+    title       CHARACTER VARYING(100)    NOT NULL UNIQUE,
```
Comment (byliuyang) [inline backend/app/adapter/sqldb/migration/29_create_incident_table.sql:5]:
```suggestion
    title       VARCHAR(150)    NOT NULL,
```
Title doesn't have to be unique.

File: backend/app/adapter/sqldb/migration/29_create_incident_table.sql [hunk 5]
```diff
@@ -0,0 +1,10 @@
...
+    created_at  TIMESTAMP WITH TIME ZONE  NOT NULL
```
Comment (byliuyang) [inline backend/app/adapter/sqldb/migration/29_create_incident_table.sql:6]:
```suggestion
    created_at  TIMESTAMP WITH TIME ZONE    NOT NULL
```

File: backend/app/adapter/sqldb/migration/30_create_progress_table.sql [hunk 6]
```diff
@@ -0,0 +1,13 @@
+-- +migrate Up
+CREATE TABLE progress
+(
+    incident_id CHARACTER VARYING(4) NOT NULL,
```
Comment (byliuyang) [inline backend/app/adapter/sqldb/migration/30_create_progress_table.sql:4]:
```suggestion
    incident_id VARCHAR(4) NOT NULL,
```

File: backend/app/adapter/sqldb/migration/30_create_progress_table.sql [hunk 7]
```diff
@@ -0,0 +1,13 @@
...
+    FOREIGN KEY (incident_id) REFERENCES "incident"(id)
+        ON UPDATE CASCADE,
```
Comment (byliuyang) [inline backend/app/adapter/sqldb/migration/30_create_progress_table.sql:6]:
```suggestion
        ON DELETE CASCADE ON UPDATE CASCADE,
```
If a incident is deleted, do we want to keep its progress? If not, please add `ON DELETE CASCADE`

File: backend/app/adapter/sqldb/migration/30_create_progress_table.sql [hunk 8]
```diff
@@ -0,0 +1,13 @@
...
+    status CHARACTER VARYING(10) NOT NULL DEFAULT 'reported',
```
Comment (byliuyang) [inline backend/app/adapter/sqldb/migration/30_create_progress_table.sql:7]:
```suggestion
    status CHARACTER VARYING(15) NOT NULL DEFAULT 'reported',
```

**Comment by byliuyang** [comment]
Let's rename the PR to `Add incident and progress DB table`

File: backend/app/adapter/sqldb/table/progress.go [hunk 9]
```diff
@@ -0,0 +1,16 @@
+package table
+
+// Progress represents database table columns for 'progress' table
+var Progress = struct {
+	TableName        string
+	ColumnIncidentId string
```
Comment (arberiii) [inline backend/app/adapter/sqldb/table/progress.go:6]:
```suggestion
	ColumnIncidentID string
```
You have to change it also here

**Pull request merged**